### DISCUSSION TITLE
Welcome to ione Discussions!

### DISCUSSION THREAD

**Discussion by ione-cloud** [ione-cloud/ione#48] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Muhammad-Aqueel** [comment]
Installing Gems:
1. Installing gems...
bundle install
rake aborted!
NoMethodError: undefined method `red' for "Command failed with status (127): [bundle install...]":String
rake/test_install.rake:17:in `fail'
rake/install_gems.rake:8:in `block in <top (required)>'
/usr/local/share/gems/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
Tasks: TOP => install => install_gems
(See full trace by running task with --trace)

I am using centos 8, opennebula 5.12, Nodejs v12.21.0 ,Ngnix but still facing this error need help to resolve it. Thank you

**Reply by slntopp** [reply]
I'll look through, shouldn't be there, but anyway, install gem colorize before running this.

`gem install colorize`

**Reply by slntopp** [reply]
If it won't work, pls send which ruby are you using and from which user are you executing that

**Reply by Muhammad-Aqueel** [reply]
> If it won't work, pls send which ruby are you using and from which user are you executing that

rake aborted!
No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:705:in raw_load_rakefile' /usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:104:in block in load_rakefile'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:186:in standard_exception_handling' /usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:103:in load_rakefile'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:82:in block in run' /usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:186:in standard_exception_handling'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:80:in run' /usr/local/share/gems/gems/rake-13.0.6/exe/rake:27:in <top (required)>'
/bin/rake:23:in load' /bin/rake:23:in

'
I installed gem colorize LINE 127 issue is resolved but above are still there.
ruby 2.5.9p229 (2021-04-05 revision 67939) [x86_64-linux] ruby version i am using. I execute
sudo rake install with oneadmin user.

**Reply by slntopp** [reply]
You're launching rake from the wrong directory, make sure you're in the directory where you cloned the repo.

Also try running `bundle install` in that dir too.

**Reply by Muhammad-Aqueel** [reply]
> You're launching rake from the wrong directory, make sure you're in the directory where you cloned the repo.
> 
> Also try running `bundle install` in that dir too.

I did both above suggested steps but errors are still there even LINE 127 error shows again.

** Invoke install_gems (first_time)
** Invoke before
** Execute install_gems
Installing Gems:
1. Installing gems...
bundle install
rake aborted!
NoMethodError: undefined method `red' for "Command failed with status (127): [bundle install...]":String
rake/test_install.rake:17:in `fail'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/file_utils.rb:67:in `block in create_shell_runner'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/file_utils.rb:57:in `sh'
rake/install_gems.rake:8:in `block in <top (required)>'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/usr/share/ruby/monitor.rb:235:in `mon_synchronize'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:243:in `block in invoke_prerequisites'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:241:in `each'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:241:in `invoke_prerequisites'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:218:in `block in invoke_with_call_chain'
/usr/share/ruby/monitor.rb:235:in `mon_synchronize'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:83:in `block in run'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/usr/local/share/gems/gems/rake-13.0.6/lib/rake/application.rb:80:in `run'
/usr/local/share/gems/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/bin/rake:23:in `load'
/bin/rake:23:in `<main>'
Tasks: TOP => install => install_gems

**Reply by slntopp** [reply]
Need more info about how do you run `rake install`, user and full log(you should hide your domain ofcourse), so I can reproduce.

**Reply by Muhammad-Aqueel** [reply]
> Need more info about how do you run `rake install`, user and full log(you should hide your domain ofcourse), so I can reproduce.

Process completed I run it with ROOT user thanks.
I don't have domain I am using public IP
Nginx going to be configured with following server names:
  cloud.X.X.X.X      --> Sunstone
  ione-api.X.X.X.X   --> IONe API
  ione-admin.X.X.X.X --> IONe UI

NOTE: X.X.X.X hiding my public IP
how can I access above 3 addresses?

**Reply by slntopp** [reply]
You can't, IONe requires domain and certificate by default.
You can install it manually and configure that later by yourself later.
Then could you please send the output of `gem list`

**Reply by Muhammad-Aqueel** [reply]
> You can't, IONe requires domain and certificate by default.
> You can install it manually and configure that later by yourself later.
> Then could you please send the output of `gem list`

*** LOCAL GEMS ***

activesupport (4.2.11.1)
addressable (2.8.0, 2.7.0)
amazon-ec2 (0.9.17)
android_key_attestation (0.3.0)
augeas (0.6.4)
awrence (1.1.1)
aws-eventstream (1.1.0)
aws-partitions (1.304.0)
aws-sdk-cloudwatch (1.36.0)
aws-sdk-core (3.94.0)
aws-sdk-ec2 (1.154.0)
aws-sdk-kms (1.30.0)
aws-sdk-s3 (1.63.0)
aws-sigv4 (1.1.3)
azure_mgmt_compute (0.19.2)
azure_mgmt_monitor (0.17.5)
azure_mgmt_network (0.23.2)
azure_mgmt_resources (0.17.8)
azure_mgmt_storage (0.21.0)
bigdecimal (1.3.4)
bindata (2.4.7)
builder (3.2.4)
bundler (2.2.24)
cbor (0.5.9.6)
chunky_png (1.3.11)
cmath (default: 1.0.0)
colorize (0.8.1)
concurrent-ruby (1.1.6)
configparser (0.1.7)
cose (0.11.0)
csv (default: 1.0.0)
curb (0.9.10)
daemons (1.4.0, 1.3.1)
dalli (2.7.10)
date (default: 1.0.0)
dbm (default: 1.0.0)
did_you_mean (1.2.0)
domain_name (0.5.20190701)
em-websocket (0.3.8)
etc (default: 1.0.0)
eventmachine (1.2.7)
faraday (0.17.3)
faraday-cookie_jar (0.0.6)
faraday_middleware (0.14.0)
fcntl (default: 1.0.0)
ffi (1.12.2)
ffi-rzmq (2.0.7)
ffi-rzmq-core (1.0.7)
fiddle (default: 1.0.0)
file-tail (1.2.0)
fileutils (default: 1.0.2)
gdbm (default: 2.0.0)
gnuplot (2.6.2)
hashie (3.6.0)
highline (1.7.10)
http-cookie (1.0.3)
i18n (0.9.5)
inflection (1.0.0)
io-console (0.4.6)
ipaddr (default: 1.2.0)
ipaddress (0.8.3)
jmespath (1.4.0)
json (2.3.0, 2.1.0)
jwt (2.2.1)
memcache-client (1.8.5)
mime-types (3.3.1)
mime-types-data (3.2020.0425)
mini_portile2 (2.4.0)
minitest (5.14.0)
ms_rest (0.7.6)
ms_rest_azure (0.11.2)
multi_json (1.15.0)
multipart-post (2.1.1)
mustermann (1.1.1)
mysql2 (0.5.3)
net-ldap (0.16.2)
net-sftp (3.0.0)
net-ssh (6.1.0)
nokogiri (1.11.7 x86_64-linux, 1.10.9)
nori (2.6.0)
openssl (2.1.2)
openssl-signature_algorithm (0.3.0)
optimist (3.0.1)
ox (2.13.2)
parse-cron (0.1.4)
pg (1.2.3)
polyglot (0.3.5)
psych (3.0.2)
public_suffix (4.0.6, 4.0.4)
racc (1.5.2)
rack (2.2.3, 2.2.2)
rack-protection (2.1.0, 2.0.8.1)
rake (13.0.6, 12.3.3)
rbvmomi (2.2.0)
rdoc (6.0.1.1)
rotp (6.0.0)
rqrcode (1.1.2)
rqrcode_core (0.1.2)
ruby2_keywords (0.0.5, 0.0.2)
safety_net_attestation (0.4.0)
scanf (default: 1.0.0)
sdbm (default: 1.0.0)
securecompare (1.0.0)
sequel (5.46.0, 5.31.0)
sinatra (2.1.0, 2.0.8.1)
sinatra-contrib (2.1.0)
sinatra-websocket (0.3.1)
sqlite3 (1.4.2)
stringio (default: 0.0.1)
strscan (default: 1.0.0)
sync (0.5.0)
thin (1.8.1, 1.7.2)
thread_safe (0.3.6)
tilt (2.0.10)
timeliness (0.3.10)
tins (1.29.1)
tpm-key_attestation (0.7.0)
treetop (1.6.10)
tzinfo (1.2.7)
unf (0.1.4)
unf_ext (0.0.7.2)
uuidtools (2.1.5)
vsphere-automation-cis (0.4.6)
vsphere-automation-runtime (0.4.6)
vsphere-automation-vcenter (0.4.6)
webauthn (2.2.0)
webrick (1.7.0, default: 1.4.2.1)
xml-simple (1.1.5)
xmlrpc (0.3.2, 0.3.0)
zendesk_api (1.25.0)
zlib (default: 1.0.0)

**Reply by Muhammad-Aqueel** [reply]
> You can't, IONe requires domain and certificate by default.
> You can install it manually and configure that later by yourself later.
> Then could you please send the output of `gem list`

It means it need to install it again after having DOMAIN. right?

**Reply by slntopp** [reply]
You'd need to configure your nginx or another webserver in order to make everything accessinble.
Basically you should have three things exposed: Sunstone(port 9869), IONe API(8009) and IONe Admin UI (static files)

**Reply by slntopp** [reply]
Also I'd need 
`which ruby`
`which bundle`
`which gem`

**Reply by Muhammad-Aqueel** [reply]
> Also I'd need
> `which ruby`
> `which bundle`
> `which gem`

which ruby
/usr/bin/ruby
which bundle
/usr/local/bin/bundle
which gem
/usr/bin/gem

**Reply by slntopp** [reply]
I'm little bit confused, what's the error now, after you manually installed colorize. If it's 
> rake aborted!
> No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)
then just ensure you're executing `rake install` in the right directory

**Reply by Muhammad-Aqueel** [reply]
> I'm little bit confused, what's the error now, after you manually installed colorize. If it's
> 
> > rake aborted!
> > No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)
> > then just ensure you're executing `rake install` in the right directory

Hello,
I execute rake install from ROOT user it worked properly but not from oneadmin USER.

**Reply by slntopp** [reply]
Installer made to be working from root only, if you need it to be working from oneadmin to, feel free to contribute or leave feature request

**Reply by Muhammad-Aqueel** [reply]
> Installer made to be working from root only, if you need it to be working from oneadmin to, feel free to contribute or leave feature request

Is there any possible way to install it to Ubuntu 18.04+ ?
Second I also tried it to install with opennebula 6.0 but problem occurs when I need to install fireedge because fireedge required NodeJS 10.X and Ione required NodeJS 12.X. Is there any solution regarding this?

**Reply by slntopp** [reply]
In automatic way - no, installer is made to work with yum only yet. We're considering updating it, but it wasn't that popular request yet. And again, feel free to contribute, the only thing needs to be changed is packages list and yum -> apt.

We haven't had any issues with NodeJS during installing IONe with OpenNebula 6.0. However if you face any, you can use something like `nvm` to manage nodejs versions. NodeJS is needed by IONe only during installation in order to build UI.
Also important to note, in most cases NodeJS is back-compatible, so NodeJS 10.x.x code should be running fine on NodeJS 12/14, depending on native libs.
